pkg/controller/certificates/sync.go

```diff
@@ -350,3 +359,35 @@ func (c *Controller) updateCertificateStatus(old, new *v1alpha1.Certificate) (*v
 	// for CRDs (https://github.com/kubernetes/kubernetes/issues/38113)
 	return c.CMClient.CertmanagerV1alpha1().Certificates(new.Namespace).Update(new)
 }
+
+func (c *Controller) calculateTimeBeforeExpiry(cert *x509.Certificate, crt *v1alpha1.Certificate) time.Duration {
+	// validate if the certificate received was with the issuer configured
+	// duration. If not we generate an event to warn the user of that fact.
+	certDuration := cert.NotAfter.Sub(cert.NotBefore)
+	if certDuration < crt.Spec.Duration.Duration {
+		s := fmt.Sprintf(messageCertificateDuration, certDuration, crt.Spec.Duration.Duration)
+		glog.Info(s)
+		c.Recorder.Event(crt, api.EventTypeNormal, infoCertificateDuration, s)
+	}
+	// renew is the duration before the certificate expiration that cert-manager
+	// will start to try renewing the certificate.
+	renew := v1alpha1.DefaultRenewBefore
+	if crt.Spec.RenewBefore.Duration != 0 {
+		renew = crt.Spec.RenewBefore.Duration
+	}
+	// Verify that the renewBefore duration is inside the certificate validity duration.
+	// If not we notify with an event that we will renew the certificate
+	// before (certificate duration / 3) of its expiration duration.
+	if renew > certDuration {
+		glog.Info(messageScheduleModified)
+		c.Recorder.Event(crt, api.EventTypeNormal, infoScheduleModified, messageScheduleModified)
```

Comment:
Same here - we need to be careful firing events. Events should only be used to signify something is started/completed, not for this sort of informational information about the status of the resource.